Makefile

```
@@ -0,0 +1,87 @@
+# btpd - BitTorrent Protocol Daemon
+# See LICENSE file for copyright and license details.
+
+BTPD_SRC    = ${wildcard btpd/*.c}
+BTPD_DEPS   = ${wildcard btpd/*.h}
+BTPD_OBJ    = ${BTPD_SRC:.c=.o}
+
+BTCLI_SRC   = ${wildcard cli/*.c}
+BTCLI_DEPS  = ${wildcard cli/*.h}
+BTCLI_OBJ   = ${BTCLI_SRC:.c=.o}
+
+BTINFO_SRC  = ${wildcard info/*.c}
+BTINFO_DEPS = ${wildcard info/*.h}
+BTINFO_OBJ  = ${BTINFO_SRC:.c=.o}
+
+MISC_SRC    = ${wildcard misc/*.c}
+MISC_DEPS   = ${wildcard misc/*.h}
+MISC_OBJ    = ${MISC_SRC:.c=.o}
+
+EVLOOP_SRC  = ${wildcard evloop/*.c}
+EVLOOP_DEPS = ${wildcard evloop/*.h}
+EVLOOP_OBJ  = ${EVLOOP_SRC:.c=.o}
+
+include config.mk
+
+all: options btpd btinfo btcli
+
+options:
+	@echo btpd build options:
+	@echo "CFLAGS   = ${CFLAGS}"
+	@echo "LDFLAGS  = ${LDFLAGS}"
+	@echo "CC       = ${CC}"
+
+.c.o:
+	${CC} -c ${DEFS} ${CPPFLAGS} ${CFLAGS} $< -o $@
+
+${%_OBJ}: ${%_DEPS}
+
+libmisc.a: ${MISC_OBJ}
+	ar rcs misc/$@ ${MISC_OBJ}
+
+libevloop.a: ${EVLOOP_OBJ}
+	ar rcs evloop/$@ ${EVLOOP_OBJ}
+
+btpd: ${BTPD_OBJ} libmisc.a libevloop.a
+	${CC} ${CFLAGS} -o btpd/$@ ${BTPD_OBJ}	 misc/libmisc.a evloop/libevloop.a ${LDFLAGS}
+
+btinfo: ${BTINFO_OBJ} libmisc.a
+	${CC} ${CFLAGS} -o info/$@ ${BTINFO_OBJ} misc/libmisc.a ${LDFLAGS}
+
+btcli: ${BTCLI_OBJ} libmisc.a
+	${CC} ${CFLAGS} -o cli/$@  ${BTCLI_OBJ}  misc/libmisc.a ${LDFLAGS}
```

I would also suggest following change, otherwise `make` followed by `make install` unnecessarily builds final binaries and archives again.

```suggestion
all: options btpd/btpd info/btinfo cli/btcli

options:
	@echo btpd build options:
	@echo "CFLAGS   = ${CFLAGS}"
	@echo "LDFLAGS  = ${LDFLAGS}"
	@echo "CC       = ${CC}"

.c.o:
	${CC} -c ${DEFS} ${CPPFLAGS} ${CFLAGS} $< -o $@

${%_OBJ}: ${%_DEPS}

misc/libmisc.a: ${MISC_OBJ}
	ar rcs $@ ${MISC_OBJ}

evloop/libevloop.a: ${EVLOOP_OBJ}
	ar rcs $@ ${EVLOOP_OBJ}

btpd/btpd: ${BTPD_OBJ} misc/libmisc.a evloop/libevloop.a
	${CC} ${CFLAGS} -o $@ ${BTPD_OBJ}	 misc/libmisc.a evloop/libevloop.a ${LDFLAGS}

info/btinfo: ${BTINFO_OBJ} misc/libmisc.a
	${CC} ${CFLAGS} -o $@ ${BTINFO_OBJ} misc/libmisc.a ${LDFLAGS}

cli/btcli: ${BTCLI_OBJ} misc/libmisc.a
	${CC} ${CFLAGS} -o $@  ${BTCLI_OBJ}  misc/libmisc.a ${LDFLAGS}
```

---

I don't made this, because i want to make more changes on the source structure, but for now this is well.